### PR TITLE
Bot API improvements

### PR DESCRIPTION
**Pull request by rasom**
Some changes that were done during hackathon and were included only in hackathon's version of app.

1. `text-message` can be used fro sending message from bot after command (if status.sendMessage is used, message will appear before command)
2. `from`/`to` props in context args for suggestions functions
3. `text-message` property and `status.sendMessage` method allows to send request from bot
4. ability to add some plain text to request message (instead of rendering command's preview )

### CODE REVIEW

File: src/status_im/chat/handlers/send_message.cljs [hunk 1]
```diff
@@ -301,7 +310,23 @@
                  current-account-id accounts contacts] :as db}
          [_ {:keys [chat-id command]}]]
       (log/debug "sending command: " command)
-      (when (not (get-in contacts [chat-id :dapp?]))
+      (if (get-in contacts [chat-id :dapp?])
+        (when-let [text-message (get-in command [:content :handler-data :text-message])]
+          (cond
```
Comment (rasom) [inline src/status_im/chat/handlers/send_message.cljs]: This code should be extracted to a separate function.

File: src/status_im/chat/handlers/input.cljs [hunk 2]
```diff
@@ -128,8 +129,12 @@
                 args    (-> (get-in db [:chats current-chat-id :input-text])
                             (input-model/split-command-args)
                             (rest))
+                to       (get-in db [:contacts current-chat-id :address])
+                from (get-in db [])
```
Comment (flexsurfer) [inline src/status_im/chat/handlers/input.cljs]: It looks like `from` is not used

File: src/status_im/chat/handlers/input.cljs [hunk 3]
```diff
@@ -128,8 +129,12 @@
...
                 params  {:parameters {:args args}
-                         :context    (merge {:data data}
+                         :context    (merge {:data data
+                                             :from       current-account-id
```
Comment (flexsurfer) [inline src/status_im/chat/handlers/input.cljs]: formatting

File: src/status_im/chat/handlers/send_message.cljs [hunk 4]
```diff
@@ -70,16 +70,17 @@
...
             request       (:request (:handler-data command))
+            handler-data  (:handler-data command)
             hidden-params (->> (:params (:command command))
                                (filter #(= (:hidden %) true))
                                (map :name))
             command'      (->> (assoc command-message :handler-data handler-data)
                                (prepare-command current-public-key chat-id clock-value request)
                                (cu/check-author-direction db chat-id))]
-        (log/debug "Handler data: " request handler-data (dissoc params :commands :command-message))
+        (log/debug "Handler data: " request (:handler-data command) (dissoc params :commands :command-message))
```
Comment (flexsurfer) [inline src/status_im/chat/handlers/send_message.cljs]: we can use here `handler-data`

File: src/status_im/utils/types.cljs [hunk 5]
```diff
@@ -13,4 +13,6 @@
 
 (defn json->clj [json]
   (when-not (= json "undefined")
-    (js->clj (.parse js/JSON json) :keywordize-keys true)))
+    (try
+      (js->clj (.parse js/JSON json) :keywordize-keys true)
+      (catch js/Error _ (when (string? json) json)))))
```
Comment (jeluard) [inline src/status_im/utils/types.cljs]: Why not check for `string?` before trying to parse?

Comment (rasom) [inline src/status_im/utils/types.cljs]: This will not help actually, even if `json` is string but has wrong format exception will be thrown.

**Comment by rasom** [comment]
@flexsurfer fixed

**Pull request merged**